### ISSUE TITLE
Coinbase advanced trading fetch_balance may be broken

### ISSUE THREAD

**Issue by HawtDogFlvrWtr** [ccxt/ccxt#16944]
### Operating System

Ubuntu 22.04.1

### Programming Languages

python, php

### CCXT Version

2.8.31

### Description

Modified your examples in python for CB advanced trading, converted to fetch_balance and it returns all values, but none of the funds that are in my account appear. I've even selected everything for the coinbase api permissions selection, to ensure it's not a permissions issue. I deposited 100$ in BTC about 2 hours ago to test. BTC isn't even listed in the return from fetch_balance(). Doesn't look like there is a v3 endpoint, so it's using the v2. It looks like the balances can be seen via advanced trading and coinbase proper on their site, so they obviously share some of the same endpoints....

TLDR; I hate coinbase

```
'APE': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AERGO': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AIOZ': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'SYN': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'HIGH': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ALICE': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'RNDR': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'QSP': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ORCA': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'FIDA': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'GLM': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ERN': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'UPI': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AVT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'SNT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'GNT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'LOOM': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'DESO': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'PLU': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'MPL': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'GFI': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'BICO': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'RBN': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'IMX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'GODS': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'STX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'DIA': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'UNFI': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'SUPER': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'POLS': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'MCO2': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'IDEX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'CTX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'BLZ': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ENS': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'SPELL': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'POWR': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'GALA': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'SHPING': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'NCT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'API3': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'PRO': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'LQTY': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'INV': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'GYEN': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'MUSD': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'FOX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'COVAL': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ALCX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'MDT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'VGX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'CRO': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'LCX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'TRAC': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'FX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'RARI': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'PERP': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'KRL': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'BADGER': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ARPA': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AVAX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AGLD': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'BTRST': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'JASMY': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'WCFG': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AUCTION': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ZEN': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'COTI': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'XYO': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'RGT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'SUKU': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'DDX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'RAD': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'TRU': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'WLUNA': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'REQ': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AXS': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'QUICK': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'UST': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'IOTX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ORN': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'TRIBE': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'RAI': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'PLA': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ASM': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'ACH': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'PAX': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'FET': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'MASK': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'FARM': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'CLV': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'RLY': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'BOND': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'QNT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'KEEP': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'CHZ': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'AMP': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'GTC': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'DOT': {'free': 0.0, 'used': 0.0, 'total': 0.0}, 'SOL': {'free': 0.0, 'used': 0.0, 'total': 0.0}
```

### Code

```
  # -*- coding: utf-8 -*-

import os
import sys
from pprint import pprint

root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(root + '/python')

import ccxt  # noqa: E402

# -----------------------------------------------------------------------------

print('CCXT Version:', ccxt.__version__)

# -----------------------------------------------------------------------------

exchange = ccxt.coinbase({
    'apiKey': 'XXXXXXX',
    'secret': 'XXXXXXXXXXX',
})


try:
    fetch_balance = exchange.fetch_balance()
    print(fetch_balance)
except Exception as err:
    print(err)

```

**Comment by HawtDogFlvrWtr** [comment]
After looking through fetch_balance function, I noticed limit : 100. After trial and error because they don't document this limit in their api documentation, I found that the max limit is 300. You're supposed to paginate somehow beyond that; but again they don't describe how best to do that. 

Anyone have an idea how to paginate beyond 300 records pulled from fetch_balance?

**Comment by carlosmiei** [comment]
Hello @HawtDogFlvrWtr you can easily paginate using the `cursor` value provided inside `['info']['pagination']`

**Comment by carlosmiei** [comment]
You can read more about it here: https://docs.cloud.coinbase.com/advanced-trade-api/reference/retailbrokerageapi_getaccounts

**Comment by pottertech** [comment]
You can use the code in the PHP examples to show you how its done.

https://github.com/ccxt/ccxt/blob/master/examples/php/coinbase-fetch-all-balances.php
